### PR TITLE
Use WMO platform naming, but strip hyphens out of ORAC filenames

### PR DESCRIPTION
**Pull request by adamcpovey**
_**This will break local scripts as it changes output file names and some driver file arguments. WE ALL NEED TO TEST THIS ONE!**_

This attempts to clean up instrument naming in ORAC. In theory, the pre and main processors will now use the same formatting for sensor/platform names while platform names will follow the WMO standard whereby there's usually a hyphen before any number. Clauses dealing with this are removed from netcdf_output_create_file.F90, read_lwrtm.F90, read_swrtm.F90 and are instead in setup.F90 when parsing input files or in sad_chan.F90 when determining the name of the text LUTs. Names for NCDF LUTs are worked out within the Python script and mostly follows the WMO standard.

An addition is made to preparation.F90 to remove hyphens from ORAC filenames as these are a protected character in CCI filenames. These are put back in by pyorac.FileName.

Theoretically, the main processor no longer requires the Ctrl%InstName
argument as it could be read from the configuration file.

### CODE REVIEW

**Comment by garethethomas** [comment]
Sensible and probably necessary change. I've tested with SLSTR and it works just fine, but note that I didn't test using your Python scripts (only a very minor modification needed for my own scripts to make it work).
I can't test with other instruments so easily, but I can try if needed.

**Review by garethethomas — APPROVED** [review]
See my previous comment...

File: pre_processing/cloud_typing_pavolonis.F90 [hunk 1]
```diff
@@ -1960,77 +1960,55 @@ function plank_inv(input_platform, T)
...
-      index = 20
-   case ("Himawari-8")
-      index = 21
-   case ("Himawari-9")
+   case ("Himawari-8", "Himawari-9")
       index = 21
-   case ("GOES-16")
-      index = 22
-   case ("GOES-17")
+   case ("GOES-16", "GOES-17")
```
Comment (simonrp84) [inline pre_processing/cloud_typing_pavolonis.F90:2005]: Could you also add "GOES-18" here?

File: pre_processing/cloud_typing_pavolonis.F90 [hunk 2]
```diff
@@ -1960,77 +1960,55 @@ function plank_inv(input_platform, T)
...
       index = 22
-   case ("SuomiNPP")
+   case ("Suomi-NPP", "NOAA-20")
```
Comment (simonrp84) [inline pre_processing/cloud_typing_pavolonis.F90:2007]: Likewise, could you add "NOAA-21" to this one.

File: pre_processing/setup.F90 [hunk 3]
```diff
@@ -840,8 +849,16 @@ subroutine setup_avhrr(args, channel_ids_user, channel_info, verbose)
...
       str1 = str1(1:j-1)
-      str2 = str2(j+1:)
-      args%platform = trim(adjustl(str2))
+      str2 = adjustl(str2(j+1:))
+      if (str2(1:4) == 'noaa') then
+         args%platform = 'NOAA-' // trim(str2(5:))
+      else if (str2(1:6) == 'metopa' .or. str2(1:7) == 'metop02') then
+         args%platform = 'Metop-A'
+      else if (str2(1:6) == 'metopb' .or. str2(1:7) == 'metop01') then
+         args%platform = 'Metop-B'
```
Comment (simonrp84) [inline pre_processing/setup.F90:858]: Could you add MetOp-C to this?
```
      else if (str2(1:6) == 'metopc' .or. str2(1:7) == 'metop03') then
         args%platform = 'Metop-C'
```

**Review by simonrp84 — COMMENTED** [review]
The PR looks fine, just a couple of minor comments to bring ORAC up-to-date with the more recent missions.

File: pre_processing/cloud_typing_pavolonis.F90 (hunk 1 above)
Comment (adamcpovey) [reply pre_processing/cloud_typing_pavolonis.F90:2005]: Does it have largely the same spectral response functions? These are subscripting coefficients to make platform X look like the NOAA that goes into the neural network.

If not, additional coefficients will need to be generated.

File: pre_processing/cloud_typing_pavolonis.F90 (hunk 2 above)
Comment (adamcpovey) [reply pre_processing/cloud_typing_pavolonis.F90:2007]: Ditto

Comment (simonrp84) [reply pre_processing/cloud_typing_pavolonis.F90:2007]: Likewise.

File: pre_processing/setup.F90 (hunk 3 above)
Comment (adamcpovey) [reply pre_processing/setup.F90:858]: Any idea which of A or B this one looks like?

**Pull request merged**